Triangulation_on_sphere_2/doc/Triangulation_on_sphere_2/Triangulation_on_sphere_2.txt

```diff
@@ -0,0 +1,241 @@
+points are sufficiently far from each other.
+
+Caroli et al. et provide bounds on this separation criterion for the case of the \f$ \mathtt{double}\f$
+number type: two points must be at least \f$ 2^{-25}r\f$ apart. This is a condition that
+is generally satisfied for most inputs: if \f$ r\f$ were for example the radius of the Earth,
+roughly 6300 kms, then the point separation requirement would be of the order of a meter.
+
+\section Section_2D_ToS_Implementation Implementation
+
+The main class of this package is the class `CGAL::Delaunay_triangulation_on_sphere_2`;
+it represents a Delaunay triangulation on the sphere, and provides insertion and removal of vertices,
+as well as tools to draw the triangulation and its dual, the Voronoi diagram.
+The base of `CGAL::Delaunay_triangulation_on_sphere_2` is the class `CGAL::Triangulation_on_sphere_2`.
+It represents a triangulation on the sphere, but does <em>not</em> support insertion or removal
+of vertices. Both classes are built on top of a data structure called the triangulation data structure.
+The triangulation data structure can be thought of as a container for the faces and vertices
+of the triangulation. This data structure also takes care of all the combinatorial aspects of the triangulation.
+
+These triangulation classes are intentionally very similar to `CGAL::Delaunay_triangulation_2` and
+`CGAL::Triangulation_2` as both classes represent triangulations of a 2-manifold domain without boundary.
+As such, many details about the implementation are not repeated here, and complementary information
+can be found in the Sections \ref Section_2D_Triangulations_Representation,
+\ref Section_2D_Triangulations_Software_Design, and \ref Section_2D_Triangulations_Basic
+of the package \ref PkgTriangulation2.
+
+However, a significant departure from Euclidean 2D triangulations is the following: since
+the triangulation data structure represents a 2-manifold without boundary, it is necessary
+for 2D triangulations to introduce so-called <i>infinite faces</i> to complete the "real"
+triangulation (see \ref Triangulation_2TheSetofFaces).
+On the sphere, this trick is not necessary as the triangulation itself is already a 2-manifold
+without boundary.
+
+\subsection Section_2D_ToS_Ghost Ghost faces
+
+There is an exception to the previous statement: in degenerate configurations where all points
+lie on the same hemisphere, the Delaunay triangulation on the sphere theoretically has a border.
+Internally however, the triangulation data structure must remain a 2-manifold at all time.
+To ensure this property, fictitious faces called <i>ghost faces</i> are added.
+These faces are characterized by the fact that the center of the sphere does not (strictly) lie
+on the positive side of the supporting plane of the face. On other hand, faces that
+are not ghost faces are called <em>solid faces</em>, and edges of such faces are <em>solid edges</em>.
```

Comment:
no need for a name for edges between ghost and solid faces?

Reply:
Those are called contour edges and there are a few functors/iterators related to them, but it is all undocumented (for no real reason, I just didn't feel like it was useful enough to document).